motor-control/tests/test_limit_switch.cpp

```diff
@@ -46,7 +46,8 @@ TEST_CASE("Move with stop condition == limit switch") {
     GIVEN("the limit switch has not been triggered") {
         test_objs.hw.set_mock_lim_sw(false);
         WHEN("the move is finished") {
-            test_objs.handler.finish_current_move();
+            while (!test_objs.handler.pulse())
```

Comment:
Generally we try to stay away from using `while()` loops, especially in tests. If you make the move short enough you should only need to call pulse like 3x times or something like that.